src/main/java/Problem1.java

```diff
@@ -1,8 +1,43 @@
-public class Problem1 {
+public class Problem1
+{
+    public static int binarySearch(int[] data, int target)
+    {
+        if(data == null || data.length == 0 || (data.length == 1 && data[0] != target))
+        {
+            return -1;
+        }
 
-    // Do not change signature (function name, parameters)
-    public static int binarySearch(int[] data, int target) {
-        // Add your solution
-        return -1; // Place holder
+        int half = (data.length - 1) / 2;
+
+        if(target == data[half])
+        {
+            return half;
+        }
+        else if(target > data[half])
+        {
+            //secondHalf starts at last value - half
+            int[] secondHalf = new int[data.length - 1 - half];
+            for(int i = 0; i < secondHalf.length; i++)
```

Comment:
for recursion solution, you shouldn't need a for loop.

Reply:
add a helper function to make it easier for recursion. call your helper from the given function.